packages/core/rfcs/20200417-products-filters.md

```diff
@@ -0,0 +1,47 @@
+# Filtering products
+
+Filtering products is an important part of any products listing page. To ensure the best developer experience and consistency between integrations (because filters have to be delivered to the theme in a unified form) we need to provide a common way to retrieve available filters.
+
+## New APIs
+
+### Modify `useProduct` composable
+
+Since there are several factors that the filters depend on (product's type, category, available products, their attributes) and the set of filters might be dynamic, this feature should introduce a new field in the `useProduct` composable, `availableFilters`, like in the example below.
+
+**New interface**:
+
+```ts
+export interface UseProduct<PRODUCT, PRODUCT_FILTER> {
+  products: ComputedProperty<PRODUCT[]>;
+  totalProducts: ComputedProperty<number>;
+  availableFilters: ComputedProperty<PRODUCT_FILTER[]>;
+  search: (params: {
+    perPage?: number;
+    page?: number;
+    sort?: any;
+    term?: any;
+    filters?: PRODUCT_FILTER[];
+    [x: string]: any;
+  }) => Promise<void>;
+  loading: ComputedProperty<boolean>;
+  [x: string]: any;
+}
+```
+
+**`UseProductFactoryParams` interface**
+
+Loading filters, their possible values, counts, etc. will require an integration-specific logic that will be part of the search function, so the factory params interface needs a new interface for search function result:
+
+```ts
+export interface ProductsSearchResult<PRODUCT, PRODUCT_FILTER> {
+  data: PRODUCT[],
+  total: number,
+  availableFilters?: PRODUCT_FILTER[];
```

Comment:
whole type should be generic (`PRODUCT_FILTERS`)

Reply:
done